### DISCUSSION TITLE
Error in calculation of Mode Grüneisen parameters

### DISCUSSION THREAD

**Discussion by zentenmiyata** [ttadano/alamode#117] [Q&A]
Dear Tadano-sensei,

 This is Miyata.  When I calculate the second and third-order interatomic force constants for two-dimensional layered materials and calculate the mode Grüneisen parameters, the resulting file is output, but I get an error message I've never seen before.

WARNING in calc_gruneisen MESSAGE: Gruneisen parameter is not real

I have tried increasing the supercell size and changing the cutoff radius of the third-order interatomic force constant, but the error persists.

What is the cause of this error message?

Best regards,
Miyata,

**Comment by ttadano** [comment]
Hi,

Could you try again by setting `MIRROR_IMAGE_CONV=1` in the `&general` field of ALM for the fitting process and see if the warning disappers? 
`MIRROR_IMAGE_CONV=1` is a hidden option, which is available in recent versions of ALAMODE (>= 1.4).

**Reply by zentenmiyata** [reply]
Dear Tadano-sensei,

Thank you for your prompt response.
I will try the method you suggested.
Unfortunately, the version of ALAMODE I am currently using is 1.3.0, and I am not confident that I will be able to update it soon.
If I get this error message, does it also cause problems with calculations other than the mode Grüneisen parameters (e.g., thermal conductivity)?

Best regards,
Miyata

**Reply by ttadano** [reply]
No, this issue is very specific to the Grüneisen parameter and does not affect thermal conductivity. Indeed, even with this warning, the Grüneisen parameter looks reasonable in most cases because the breakdown of the permutation symmetry (which is the cause of the observed nonreality) is usually very small.

**Reply by zentenmiyata** [reply]
Dear Tadano-sensei,

Thank you very much for your prompt response.
I understand that the problem is specific to the mode Grüneisen parameter.
I will update the version and try the method you described to see if it solves the problem.

Best regards,
Miyata,

**Reply by zentenmiyata** [reply]
Dear Tadano-sensei,

 This is Miyata. I tried again by setting MIRROR_IMAGE_CONV=1 in the &general field of ALM ver 1.4.2 for the fitting process and see if the warning disappers, however, it was remaining.

The crystal symmetry is slightly broken during structural optimization. Is this the cause?

Best regards,
Miyata,

**Reply by ttadano** [reply]
> The crystal symmetry is slightly broken during structural optimization. Is this the cause?

Could be. You can check this point easily by using a symmetrized structure for ALM and ANPHON. You don't need to redo the force calculations. Just simply use the `DFSET` you already have.

For symmetrizing the structure, I would recommend using `spglib`.

**Reply by zentenmiyata** [reply]
Dear Tadano-sensei,

Thank you very much for your prompt response.

I was analyzing the symmetry in spglib and found that the symmetry is found for TOLERANCE = 10-3, but below that, the symmetry changes and the symmetry is slightly broken. Therefore, I think that the warning message in the Grüneisen parameter is due to the broken symmetry of the crystals.

I will try to optimize the structure while keeping the symmetry by tightening the constraints.

Best regards,
Miyata,
